### PR TITLE
fix: mobile event types and avatars

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Mobile event types needed some love, empty personal event types now show an empty state, and also some subteam avatars were not loading for orgs.

| Before | After |
|--------|--------|
| <kbd></kbd> | <kbd></kbd> |
| <kbd></kbd> | image |

Same applies for desktop:

| Before | After |
|--------|--------|
| <kbd></kbd> | <kbd></kbd> |

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Open event types page and shrink the window to get mobile treatment and see how it behaves like the pictures.

### CODE REVIEW

File: apps/web/pages/event-types/index.tsx [hunk 1]
```diff
@@ -108,22 +108,26 @@ const MobileTeamsTab: FC<MobileTeamsTabProps> = (props) => {
   const orgBranding = useOrgBranding();
   const tabs = eventTypeGroups.map((item) => ({
     name: item.profile.name ?? "",
-    href: item.teamId ? `/event-types?teamId=${item.teamId}` : "/event-types",
-    avatar: item.profile.image ?? `${orgBranding?.fullDomain ?? WEBAPP_URL}/${item.profile.slug}/avatar.png`,
+    href: item.teamId ? `/event-types?teamId=${item.teamId}` : "/event-types?noTeam",
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:111]: We need to differentiate a team pathname+query from personal account so the horizontal tabs can correctly identify the active tab

File: apps/web/pages/event-types/index.tsx [hunk 2]
```diff
@@ -108,22 +108,26 @@ const MobileTeamsTab: FC<MobileTeamsTabProps> = (props) => {
...
+    avatar: orgBranding
+      ? `${orgBranding.fullDomain}${item.teamId ? "/team" : ""}/${item.profile.slug}/avatar.png`
+      : item.profile.image ?? `${WEBAPP_URL + (item.teamId && "/team")}/${item.profile.slug}/avatar.png`,
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:114]: This was preventing org subteams to render correctly

File: apps/web/pages/event-types/index.tsx [hunk 3]
```diff
@@ -108,22 +108,26 @@ const MobileTeamsTab: FC<MobileTeamsTabProps> = (props) => {
...
   }));
   const { data } = useTypedQuery(querySchema);
   const events = eventTypeGroups.filter((item) => item.teamId === data.teamId);
 
   return (
     <div>
       <HorizontalTabs tabs={tabs} />
-      {events.length && (
+      {events.length > 0 ? (
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:122]: This was showing a "0" in the UI

File: apps/web/pages/event-types/index.tsx [hunk 4]
```diff
@@ -108,22 +108,26 @@ const MobileTeamsTab: FC<MobileTeamsTabProps> = (props) => {
...
         <EventTypeList
           types={events[0].eventTypes}
           group={events[0]}
           groupIndex={0}
           readOnly={events[0].metadata.readOnly}
         />
+      ) : (
+        <CreateFirstEventTypeView />
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:130]: We want to show an empty screen

File: apps/web/pages/event-types/index.tsx [hunk 5]
```diff
@@ -358,6 +362,10 @@ export const EventTypeList = ({ group, groupIndex, readOnly, types }: EventTypeL
     }
   }, []);
 
+  if (!types.length) {
+    return group.teamId ? <EmptyEventTypeList group={group} /> : <CreateFirstEventTypeView />;
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:366]: To show correct empty screen

File: apps/web/pages/event-types/index.tsx [hunk 6]
```diff
@@ -712,7 +720,10 @@ const EventTypeListHeading = ({
       <Avatar
         alt={profile?.name || ""}
         href={teamId ? `/settings/teams/${teamId}/profile` : "/settings/my-account/profile"}
-        imageSrc={`${orgBranding?.fullDomain ?? WEBAPP_URL}/${profile.slug}/avatar.png` || undefined}
+        imageSrc={
+          `${orgBranding?.fullDomain ?? WEBAPP_URL}${teamId ? "/team" : ""}/${profile.slug}/avatar.png` ||
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:724]: Somehow we were missing the "/team" part of the avatar URL to correctly show a team or user avatar for orgs

File: apps/web/pages/event-types/index.tsx [hunk 7]
```diff
@@ -759,6 +770,12 @@ const CreateFirstEventTypeView = () => {
       Icon={LinkIcon}
       headline={t("new_event_type_heading")}
       description={t("new_event_type_description")}
+      className="mb-16"
+      buttonRaw={
+        <Button href="?dialog=new" variant="button">
+          {t("create")}
+        </Button>
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:777]: Added the "create" button to simplify the action for mobile users mainly

File: apps/web/pages/event-types/index.tsx [hunk 8]
```diff
@@ -893,8 +910,10 @@ const Main = ({
                     groupIndex={index}
                     readOnly={group.metadata.readOnly}
                   />
-                ) : (
+                ) : group.teamId ? (
                   <EmptyEventTypeList group={group} />
+                ) : (
+                  <CreateFirstEventTypeView />
```
Comment (leog) [inline apps/web/pages/event-types/index.tsx:916]: Empty screen FTW

File: apps/web/pages/signup.tsx [hunk 9]
```diff
@@ -142,7 +142,7 @@ export default function Signup({ prepopulateFormValues, token, orgSlug }: Signup
                   <TextField
                     addOnLeading={
                       orgSlug
-                        ? getOrgFullDomain(orgSlug, { protocol: true })
+                        ? `${getOrgFullDomain(orgSlug, { protocol: true })}/`
```
Comment (leog) [inline apps/web/pages/signup.tsx:145]: Missing trailing "/" on invite

File: packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts [hunk 10]
```diff
@@ -275,8 +275,7 @@ export const getByViewerHandler = async ({ ctx, input }: GetByViewerOptions) =>
 
   const bookerUrl = await getBookerUrl(user);
   return {
-    // don't display event teams without event types,
-    eventTypeGroups: eventTypeGroups.filter((groupBy) => groupBy.parentId || !!groupBy.eventTypes?.length),
+    eventTypeGroups,
```
Comment (leog) [inline packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts:278]: We now show any event type group even if it is empty, as we show empty screens for them now

File: packages/ui/components/empty-screen/EmptyScreen.tsx [hunk 11]
```diff
@@ -48,7 +48,13 @@ export function EmptyScreen({
...
         )}
         <div className="flex max-w-[420px] flex-col items-center">
-          <h2 className="text-semibold font-cal text-emphasis mt-6 text-center text-xl">{headline}</h2>
+          <h2
+            className={classNames(
+              "text-semibold font-cal text-emphasis text-center text-xl",
+              Icon && "mt-6"
+            )}>
+            {headline}
+          </h2>
```
Comment (leog) [inline packages/ui/components/empty-screen/EmptyScreen.tsx:57]: When no icon is present and no description is passed, there was an extra margin on the title that didn't look good

**Review by PeerRich — APPROVED** [review]
thank you!!

**Pull request merged**